tests/handlers/test_agent.py

```diff
@@ -30,3 +36,37 @@ def test_post_agent_unauthenticated(test_client, db_session):
 
     assert response.status_code == 403
     assert response.json()['detail'] == "Permission denied."
+
+
+def test_list_agent_empty_list(db_session, test_client, session_cookie):
+    response = test_client.get('/agent', cookies=session_cookie)
+
+    assert response.json() == []
+
+
+def test_list_agent_success(db_session, test_client, session_cookie):
+    agent_id_1 = uuid.uuid4()
+    agent_id_2 = uuid.uuid4()
+    agent = Agent(id=agent_id_1, name='test',
+                  oauth_client=OAuthClient(type='confidential'))
+    agent_2 = Agent(id=agent_id_2, name='test2',
+                    oauth_client=OAuthClient(type='confidential'))
+    db_session.add(agent)
+    db_session.add(agent_2)
+    db_session.commit()
+
+    response = test_client.get('/agent', cookies=session_cookie)
+
+    assert response.status_code == 200
```

Comment:
```
    assert response.status_code == 200
    assert len(response_body) == 2

    assert agent_data['name'] in ['test', 'test2']
    assert agent_data['id'] in [str(agent_id_1), str(agent_id_2)]
    assert agent_data['connection_state'] == 'disconnected'
```

Ideally we'd assert the connection state further but that'll get us working with the `WebSocketManager` again.